packages/optimistic-oracle/src/proposer.js

```diff
@@ -141,14 +149,11 @@ class OptimisticOracleProposer {
       message: "Checking for proposals and disputes to settle",
     });
 
-    const settleableAccounts = [this.account, ...this.otherAccountsToSettle];
-    const settleableRequests = settleableAccounts
-      .map((account) =>
-        this.optimisticOracleClient
-          .getSettleableProposals(account)
-          .concat(this.optimisticOracleClient.getSettleableDisputes(account))
-      )
-      .flat();
+    // If settleAllRequests is true, pass undefined, so it settles _any_ expired requests.
```

Comment:
this is such a JS way of doing something by overloading the undefined case. TX would not have been happy with you here haha.

Reply:
Yeeeep. I do wish JS had true overloading for stuff like this. Would make things a lot easier.

Reply:
In compiled languages, it would just swap out the implementation at compile-time depending on the type being passed in.